halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/model/v1/plugins/Plugin.java

```diff
@@ -72,4 +85,47 @@ public InputStream downloadManifest() {
               .build());
     }
   }
+
+  public static Map<String, Object> normalizeOptions(Map<String, Object> opts) {
+    Map<String, Object> result = new HashMap<>();
+    for (Map.Entry<String, Object> entry : opts.entrySet()) {
+      String key = entry.getKey();
+      result.putAll(parseOption(key, entry.getValue()));
+    }
+
+    return result;
+  }
+
+  public static Map<String, Object> merge(Map original, Map newMap) {
+    for (Object key : newMap.keySet()) {
+      if (newMap.get(key) instanceof Map && original.get(key) instanceof Map) {
+        Map originalChild = (Map) original.get(key);
+        Map newChild = (Map) newMap.get(key);
+        original.put(key, Plugin.merge(originalChild, newChild));
+      } else if (newMap.get(key) instanceof List && original.get(key) instanceof List) {
+        List originalChild = (List) original.get(key);
+        List newChild = (List) newMap.get(key);
+        for (Object each : newChild) {
```

Comment:
`originalChild.addAll(newChild)`
Note that this is not how Spring works (it replaces the array) and I think this would be a better choice.

Reply:
ohhhh interesting, so spring just overwrites one array with another one. What is the spring way of doing it? eg. just `original.merge(newMap)`?